simpleclient_eclipselink/src/main/java/io/prometheus/client/eclipselink/EclipseLinkStatisticsCollector.java

```diff
@@ -0,0 +1,744 @@
+                if (value != null) {
+                  return ((Long) value).doubleValue();
+                }
+                return 0L;
+              }
+            }
+        ),
+        createCounter(
+            "eclipselink_connects_total",
+            "Total number of connects",
+            new ValueProvider() {
+              @Override
+              public double getValue(Map<String, Object> metrics) {
+                Object value = metrics.get(SessionProfiler.Connects);
+                if (value != null) {
+                  return ((Long) value).doubleValue();
+                }
+                return 0L;
+              }
+            }
+        )
+        ,
+        createCounter(
+            "eclipselink_disconnects_total",
+            "Total number of disconnects",
+            new ValueProvider() {
+              @Override
+              public double getValue(Map<String, Object> metrics) {
+                Object value = metrics.get(SessionProfiler.Disconnects);
+                if (value != null) {
+                  return ((Integer) value).doubleValue();
+                }
+                return 0;
+              }
+            }
+
+        ),
+        createCounter(
+            "eclipselink_optimistic_lock_exceptions_total",
+            "Total number of optimistic lock exceptions",
+            new ValueProvider() {
+              @Override
+              public double getValue(Map<String, Object> metrics) {
+                Object value = metrics.get(SessionProfiler.OptimisticLockException);
+                if (value != null) {
+                  return ((Integer) value).doubleValue();
+                }
+                return 0;
+              }
+            }
+        ),
+        createCounter(
+            "eclipselink_rcm_received_total",
+            "Total number of messages received",
+            new ValueProvider() {
+              @Override
+              public double getValue(Map<String, Object> metrics) {
+                Object value = metrics.get(SessionProfiler.RcmReceived);
+                if (value != null) {
+                  return ((Integer) value).doubleValue();
+                }
+                return 0;
+              }
+            }
+        ),
+        createCounter(
+            "eclipselink_rcm_sent_total",
+            "Total number of sent messages",
+            new ValueProvider() {
+              @Override
+              public double getValue(Map<String, Object> metrics) {
+                Object value = metrics.get(SessionProfiler.RcmSent);
+                if (value != null) {
+                  return ((Integer) value).doubleValue();
+                }
+                return 0;
+              }
+            }
+        )
+    );
+  }
+
+  private List<MetricFamilySamples> getTimers() {
+    return Arrays.<MetricFamilySamples>asList(
+        createGauge(
+            "eclipselink_uow_commits_seconds",
```

Comment:
These all sound like counters

Reply:
They all measure durations of particular operations in seconds, not sure how they can be counters. As durations, they can arbitrarily go up and down.

Reply:
In that case see if you can get to the underlying data, and create a Summary from that

Reply:
I see your point, but EclipseLink API returns one value at a time which represents the cumulative duration of operation. Unfortunately, I don't see how I can get enough data for a summary or a histrogram in one scrape.